akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala

```diff
@@ -22,38 +24,48 @@ class FramingSpec extends AkkaSpec {
   val settings = ActorMaterializerSettings(system)
   implicit val materializer = ActorMaterializer(settings)
 
-  class Rechunker extends PushPullStage[ByteString, ByteString] {
-    private var rechunkBuffer = ByteString.empty
+  class Rechunker extends GraphStage[FlowShape[ByteString, ByteString]] {
 
-    override def onPush(chunk: ByteString, ctx: Context[ByteString]): SyncDirective = {
-      rechunkBuffer ++= chunk
-      rechunk(ctx)
-    }
+    val out: Outlet[ByteString] = Outlet("Rechunker.out")
+    val in: Inlet[ByteString] = Inlet("Rechunker.in")
+    override val shape: FlowShape[ByteString, ByteString] = FlowShape(in, out)
 
-    override def onPull(ctx: Context[ByteString]): SyncDirective = {
-      rechunk(ctx)
-    }
+    override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+      new GraphStageLogic(shape) with InHandler with OutHandler {
```

Comment:
You need to `setHandlers` also. This is why `akka.stream.scaladsl.FramingSpec` is failing.